src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +46,42 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException, ParseException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateFromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToDate = LocalDate.parse(dateTo, formatter);
+        String[] parsedData = new String[4];
+        int[] salary = new int[names.length];
+
+        if (dateFromDate.isAfter(dateToDate)) {
+            throw new IllegalDateParametersException();
+        }
+
+        for (int i = 0; i < data.length; i++) {
+            parsedData = data[i].split(" ");
+            LocalDate dateTemp = null;
+            dateTemp = LocalDate.parse(parsedData[0], formatter);
+            for (int j = 0; j < names.length; j++) {
+                if (names[j].equals(parsedData[1])) {
+                    if (dateTemp.equals(dateFromDate) || dateTemp.equals(dateToDate)
+                            || dateTemp.isAfter(dateFromDate) && dateTemp.isBefore(dateToDate)) {
+                        salary[j] = salary[j] + (Integer.parseInt(parsedData[2])
+                                * Integer.parseInt(parsedData[3]));
+                    }
+                }
+            }
+        }
+
+        return buildReport(names, salary, dateFrom, dateTo);
+    }
+
+    private String buildReport(String[] names, int[] salary, String dateFrom, String dateTo) {
+        StringBuilder report = new StringBuilder();
+        report.append("Отчёт за период ").append(dateFrom).append(" - ")
+                .append(dateTo).append("\n");
+        for (int i = 0; i < names.length; i++) {
```

Comment:
try to create solution without this method. hint: you can build a report while processing data